test/cace/test_amm_semtype_cnst.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2011-2024 The Johns Hopkins University Applied Physics
+ * Laboratory LLC.
+ *
+ * This file is part of the Delay-Tolerant Networking Management
+ * Architecture (DTNMA) Tools package.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cace/amm/semtype_cnst.h>
+#include <cace/ari/text_util.h>
+#include <cace/ari/cbor.h>
+#include <cace/util/logging.h>
+#include <unity.h>
+
+// Allow this macro
+#define TEST_CASE(...)
+
+void suiteSetUp(void)
+{
+    cace_openlog();
+}
+
+int suiteTearDown(int failures)
+{
+    cace_closelog();
+    return failures;
+}
+
+static void check_cnst(const amm_semtype_cnst_t *cnst, const char *inhex, bool expect)
+{
+    string_t intext;
+    string_init_set_str(intext, inhex);
+    cace_data_t indata;
+    cace_data_init(&indata);
+    int res = base16_decode(&indata, intext);
+    string_clear(intext);
+    TEST_ASSERT_EQUAL_INT_MESSAGE(0, res, "base16_decode() failed");
+
+    ari_t val = ARI_INIT_UNDEFINED;
+    res       = ari_cbor_decode(&val, &indata, NULL, NULL);
+    cace_data_deinit(&indata);
+    TEST_ASSERT_EQUAL_INT_MESSAGE(0, res, "ari_cbor_decode() failed");
+
+    res = amm_semtype_cnst_is_valid(cnst, &val);
+    TEST_ASSERT_EQUAL_INT_MESSAGE(expect, res, "amm_semtype_cnst_is_valid() failed");
+
+    ari_deinit(&val);
+}
+
+TEST_CASE("F7", false)             // ari:undefined
```

Comment:
**Question**: Are these tests using CBOR since that is more likely to be used in fielded systems? Or should we consider using text for unit tests going forward?

Reply:
Within the C implementation the text decoding is actually optional (depends on FLEX+BISON) so only the binary form is guaranteed to be readable on all CACE configs.